scripts/git/prune_old_tags.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Deletes the input number of tags and corresponding releases, starting with the
+# oldest tags.
+
+DELETE_TAGS_COUNT="$1"
```

Comment:
I'd probably choose either a number of tags to _keep_ (e.g. 100) or a date after which to delete (3 months old), rather than a number to delete. With a number to delete, it would be easy to delete all tags without meaning to (delete 100 when only 50 exist).

Reply:
Conveniently, the tag name format already contains the date in a lexicographically-sortable format. YOu can pipe the tag names to awk to filter in this way. Something like: `| awk -F'[.-]' '$2 < 20210601'`

Reply:
Changed the script to keep the input number of tags rather than deleting them